File: caesarCipher/JS/cipher.js
const caesarCipher = (string, encoding) => {
  let result = '';
  const alpha = 'abcdefghijklmnopqrstuvwxyz';
  const k = encoding > 26 ? encoding % 26 : encoding;
  const upper = alpha.toUpperCase();

  const decoded = alpha.substring(k) + alpha.substring(0, k);
  const decodUpper = upper.substring(k) + upper.substring(0, k);

  for (let i = 0; i < string.length; i++) {
    if (alpha.includes(string[i])) {
      result += decoded[alpha.indexOf(string[i])];
    } else if (upper.includes(string[i])) {
      result += decodUpper[upper.indexOf(string[i])];
    } else {
      result += string[i];
    }
  }

  return result;
}

module.exports = caesarCipher;

File: caesarCipher/Java/Cipher.java
public class Cipher {
    public static String caesarCipher(String s, int k) {
        if (k / 26 > 0) k = k % 26;
        String alphabet = "abcdefghijklmnopqrstuvwxyz";
        String upper = alphabet.toUpperCase();
        String encoded = alphabet.substring(k, 26) + alphabet.substring(0, k);
        String encodedUpper = encoded.toUpperCase();

        StringBuilder result = new StringBuilder("");

        for (int i = 0; i < s.length(); i++) {
            if (alphabet.indexOf(s.charAt(i)) != -1) {
                result.append(encoded.charAt(alphabet.indexOf(s.charAt(i))));
            } else if (upper.indexOf(s.charAt(i)) != -1) {
                result.append(encodedUpper.charAt(upper.indexOf(s.charAt(i))));
            } else {
                result.append(s.charAt(i));
            }
        }

        return result.toString();
    }
}
